Added api to have access on the Dialog view object on the router level, when needed. Closes #263.

Nublo opened the pull request:
**Description**:
Update DialogLauncher api to have lambda which is notified when the dialog view is created.

**Related issue(s)**: #263

zsoltk commented on libraries/rib-base/src/main/java/com/badoo/ribs/android/dialog/DialogLauncher.kt:
```diff
@@ -1,8 +1,10 @@
 interface DialogLauncher {
 
-    fun show(dialog: Dialog<*>, onClose: () -> Unit)
+    fun show(dialog: Dialog<*>, onClose: () -> Unit, onShown: (AlertDialog) -> Unit)
```
Reorder + rename for consistency?

```suggestion
    fun show(dialog: Dialog<*>, onShow: (AlertDialog) -> Unit, onClose: () -> Unit)
```

CherryPerry replied:
We should be **very** careful with migrations then.
If we have `show(dialog) { someCode() }` then it will change current behaviour.

zsoltk commented on libraries/rib-base/src/main/java/com/badoo/ribs/android/dialog/DialogLauncher.kt:
```diff
@@ -1,8 +1,10 @@
 package com.badoo.ribs.android.dialog
 
+import androidx.appcompat.app.AlertDialog
```
I don't like that we'll depend on AlertDialog in this more generic interface.

Nublo commented:
Updated implementation details to immediately attach the dialog when creating AlertDialogBuilder.
This solves the described issue.

Also updated logic on managing dialogs to avoid memory leaks.

zsoltk commented on libraries/rib-base/src/main/java/com/badoo/ribs/android/dialog/DialogExtensions.kt:
```diff
@@ -45,8 +46,14 @@ private fun AlertDialog.Builder.setRib(dialog: Dialog<*>, context: Context) {
+                // java.lang.IllegalStateException: The specified child already has a parent.
+                // Detach node here -> no harmful effect when view is not attached
+                // PushTwoPopOneDefaultTest.noPermanent_singleInitial_pushOverlay_pushContent_pop - to reproduce
+                host.detachChild(it.node)
```
If this is needed to make it pass, the problem is probably somewhere else, and should be fixed there?

Nublo replied:
I am not sure, in theory it can also happen in production environment.
Here are more details on why it's happening

```
[configurationDialog] push
AlertDialog.show()
[configuration] push
AlertDialog.dismiss() -> Here we starting dismissing the dialog after new configuration
[configuration] pop -> But we immediately popping new config
AlertDialog.show() -> And show dialog again without detaching the view from host
onDismissListener
AlertDialog.dismiss()
onDismissListener
```

CherryPerry commented on libraries/rib-base/src/androidTest/java/com/badoo/ribs/android/dialogkeyboard/AlertDialogKeyboardTest.kt:
```diff
@@ -0,0 +1,79 @@
+    @Test
+    fun keyboardIsShownForRIBDialogWithEditText() {
+        setupRibs()
+        onView(withClassName(endsWith("EditText"))).perform(click())
```
Why do you need a click, when keyboard should be displayed automatically?

Nublo replied:
Keyboard will not be shown automatically if the view doesn't have focus.
I think that it's safer to make click here comparing to requesting focus in some other place.

The test is checking that the keyboard will be shown and not blocked from presenting, but not that keyboard will be shown automatically.
I will update test name to avoid confusions.
